airbyte-integrations/connectors/source-amazon-ads/unit_tests/test_streams.py

```diff
@@ -0,0 +1,285 @@
+    profiles_response = loads(profiles_response)
+    for profile in profiles_response:
+        profile["accountInfo"]["type"] = "vendor"
+    profiles_response = dumps(profiles_response)
+    setup_responses(profiles_response=profiles_response, campaigns_response=campaigns_response)
+
+    source = SourceAmazonAds()
+    streams = source.streams(test_config)
+    profile_stream = streams[0]
+    campaigns_stream = streams[1]
+    records = profile_stream.read_records(SyncMode.full_refresh)
+    profile_records = [r for r in records]
+    records = campaigns_stream.read_records(SyncMode.full_refresh)
+    campaigns_records = [r for r in records]
+    assert len(campaigns_records) == len(profile_records) * len(loads(campaigns_response))
+
+
+@pytest.mark.parametrize(
+    ("page_size"),
+    [1, 2, 5, 1000000],
+)
+@responses.activate
+def test_streams_campaigns_pagination(mocker, test_config, profiles_response, campaigns_response, page_size):
+    mocker.patch("source_amazon_ads.streams.common.PaginationStream.page_size", page_size)
+    profiles_response = loads(profiles_response)
+    for profile in profiles_response:
+        profile["accountInfo"]["type"] = "vendor"
+    profiles_response = dumps(profiles_response)
+    setup_responses(profiles_response=profiles_response)
+
+    source = SourceAmazonAds()
+    streams = source.streams(test_config)
+    campaigns_stream = streams[1]
+    profile_stream = streams[0]
+    campaigns = loads(campaigns_response)
+
+    def campaigns_paginated_response_cb(request):
+        query = urlparse(request.url).query
+        query = parse_qs(query)
+        start_index, count = (int(query[f][0]) for f in ["startIndex", "count"])
+        response_body = campaigns[start_index : start_index + count]
+        return (200, {}, dumps(response_body))
+
+    responses.add_callback(
+        responses.GET,
+        "https://advertising-api.amazon.com/sd/campaigns",
+        content_type="application/json",
+        callback=campaigns_paginated_response_cb,
+    )
+    profile_records = profile_stream.read_records(SyncMode.full_refresh)
+    profile_records = [r for r in profile_records]
+
+    campaigns_records = campaigns_stream.read_records(SyncMode.full_refresh)
+    campaigns_records = [r for r in campaigns_records]
+    assert len(campaigns_records) == len(profile_records) * len(loads(campaigns_response))
+
+
+@pytest.mark.parametrize(("status_code"), [HTTPStatus.FORBIDDEN, HTTPStatus.UNAUTHORIZED])
+@responses.activate
+def test_streams_campaigns_pagination_403_error(mocker, status_code, test_config, profiles_response, campaigns_response):
+    setup_responses(
+        profiles_response=profiles_response,
+    )
+    responses.add(responses.GET, "https://advertising-api.amazon.com/sd/campaigns", json={"message": "msg"}, status=status_code)
+    source = SourceAmazonAds()
+    streams = source.streams(test_config)
+    profile_stream = streams[0]
```

Comment:
Above I described `get_stream_by_name` function. Please, upgrade all such code (like `streams[0]`, `streams[1]` etc) to use `get_stream_by_name` function because again if an order of the streams will be changed then there will be a lot of places which will require fixing.

Reply:
Updated